native/native_main.py

```diff
@@ -153,7 +158,12 @@ def handleMessage(message):
             file.write(message["content"])
 
     elif cmd == 'temp':
-        (handle, filepath) = tempfile.mkstemp()
+        prefix = message.get('hostname')
```

Comment:
I agree with making it more general, but left to my own devices I would be tempted to leave the sanitizing in `native_main.py`, since this is the code that is actually opening the file.